spline_background.py

```diff
@@ -1,46 +1,53 @@
 import numpy as np
 from scipy import interpolate
 
+import scipp as sc
 
-def bspline_background(variable, x_values, dim=None):
+
+def bspline_background(variable, dim):
     """
     Use scipy bspline method to create a fitting function to data in `variable`.
     Knots and splines are evaluated internally.
 
     Parameters
     ----------
-        variable: scipp variable
-            The variable to which the spline should be applied
-        x_values: nupmy array
-            X values for the variable data
+        variable: scipp DataArray
+            DataArray container to which values the spline should be applied
         dim: scipp Dim
             The dimension along which the spline should be calculated
+        output: scipp DataArray
+            DataArray container with the spline
     """
-
     if dim is None:
         raise ValueError("bspline_background: dimension must be specified.")
+    if not isinstance(dim, sc.Dim):
+        raise ValueError("bspline_background: dimension must be of Dim type.")
 
-    values = variable[dim, :].values
-    errors = variable[dim, :].variances
-    weights = [1.0/i*i for i in errors]
+    x_values = variable.coords[dim].values
+    values = variable.values
+    errors = variable.variances
+    weights = 1.0/errors
 
     # spline smoothing from
     # https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.UnivariateSpline.html
     s = len(values)*np.std(errors)*np.std(errors)
     # find out the knots and splines.
     knots, u = interpolate.splprep([x_values, values], s=s, k=5, w=weights)
     # perform the B-spline interpolation
-    splined_variable = interpolate.splev(u, knots)
-    # splined_variable is a [2][data_length] array for X and Y values
+    splined = interpolate.splev(u, knots)
+    # splined is a [2][data_length] array for X and Y values
+
+    # cast splined into DataArray type
+    output_x = sc.Variable(dims=[sc.Dim.X], values=splined[0])
+    output_y = sc.Variable(dims=[sc.Dim.X], values=splined[1])
+    output_data = sc.DataArray(data=output_y, coords={sc.Dim.X: output_x})
```

Comment:
The output data has no uncertainties, is this intentional? As far as I can tell, the Mantid version does set errors: https://github.com/mantidproject/mantid/blob/13e81d9e68f084277f15dbb6bcebb8636d25f58a/Framework/CurveFitting/src/Algorithms/SplineBackground.cpp#L268-L272.

Reply:
It should be included. It is now